### PR TITLE
Feature/add network

### PR DESCRIPTION
**Pull request by Yamaguchi**
- Add `network-id` parameter to calculate message magic
- Use malfix_txid method insted of txid method
- Remove Bitcoin version check

### CODE REVIEW

File: src/chain.rs [hunk 1]
```diff
@@ -59,3 +53,71 @@ impl From<BNetwork> for Network {
         }
     }
 }
+
+/// Network ID is identifier of the Tapyrus network
+#[derive(Clone, Debug, PartialEq)]
+pub struct NetworkId(u32);
```
Comment (azuchi) [inline src/chain.rs:59]: `Network` and `NetworkId` are separated, but isn't it better for the structure to have the mode(prod or dev) and id in the `Network` struct?

File: src/chain.rs [hunk 2]
```diff
@@ -15,109 +16,60 @@ lazy_static! {
 }
 
 #[derive(Debug, Copy, Clone, PartialEq, Hash, Serialize, Ord, PartialOrd, Eq)]
-pub enum Network {
+pub struct Network {
+    pub network_type: NetworkType,
```
Comment (azuchi) [inline src/chain.rs:20]: How abount only `type` instead of `network_type`?

Comment (Yamaguchi) [reply src/chain.rs:20]: `type` is a reserved word, so using `type` generates a build error.

Comment (azuchi) [reply src/chain.rs:20]: I see.

**Review by azuchi — APPROVED** [review]

**Pull request merged**